butlarr/services/bazarr.py

```diff
@@ -0,0 +1,313 @@
+                status
+            ), "Could not reach compatible api. Is the service down? Is your API key correct?"
+            api_version = status.get("data", {}).get("bazarr_version", "")
+            assert api_version, "Could not find compatible api."
+            return api_version
+    
+    
+    def search(self, arr_variant, id):
+        if arr_variant == ArrVariant.RADARR:
+            status = self.request('providers/movies', params={'radarrid': id}, fallback=[])
+            return status.get('data') if len(status) > 0 else status
+        if arr_variant == ArrVariant.SONARR:
+            status = self.request('providers/episodes', params={'episodeid': id}, fallback=[])
+            return status.get('data') if len(status) > 0 else status
+        else:
+            assert False, f"Bazarr integration not Implemented"
+
+    def download(
+        self,
+        service,
+        id,
+        item=None,
+        options={},
+    ):
+        assert item, "Missing required arg! You need to provide a item!"
+
+        if service == ArrVariant.RADARR:
+            
+            method = 'providers/movies'
+            params = {
+                "radarrid": id, 
+                "hi": item.get('hearing_impaired'), 
+                "forced": item.get('forced'),
+                "original_format": item.get('original_format'),
+                "provider": item.get('provider'),
+                "subtitle": item.get('subtitle'),
+                **options
+            }
+
+        elif service == ArrVariant.SONARR:
+            
+            method = 'providers/episodes'
+            params = {
+                "seriesid": item.get('id'),
+                "episodeid": id,
+                "hi": item.get('hearing_impaired'), 
+                "forced": item.get('forced'),
+                "original_format": item.get('original_format'),
+                "provider": item.get('provider'),
+                "subtitle": item.get('subtitle'),
+                **options
+            }
+
+        else:
+            logger.error(
+                f"Bazarr integration with service {service} is Not Implemented"
+            )
+            return False
+        
+        return self.request(
+                method,
+                action=Action.POST,
+                params=params,
+                raw=True
+            )
+
+    
+    def create_message(self, state: State, full_redraw=False, allow_edit=False):
+        if not state.items:
+            keyboard_markup = self.keyboard(state, allow_edit=allow_edit)
+
+            return Response(
+                caption='',
+                reply_markup=keyboard_markup,
+                state=state,
+            )
+        
+        item = state.items[state.index]
+
+        parent_service = state.parent.service
+        parent_state = state.parent.state
+        media_item = parent_state.items[parent_state.index]
+
+        if ArrVariant(parent_service.arr_variant) == ArrVariant.SONARR:
```

Comment:
Same thing here, why are we doing this service specific logic here and not have a common function with an interface given by Addon

Reply:
Same answer here :)